### PR TITLE
luci-app-https_dns_proxy: remove dnsmasq integration, better service controls, providers as plugins

### PR DESCRIPTION
**Pull request by stangri**
Signed-off-by: Stan Grishin <stangri@melmac.net>

As dnsmasq integration is moved to the principal service in https://github.com/openwrt/packages/pull/10413, luci app no longer needs that functionality.

### CODE REVIEW

File: applications/luci-app-https_dns_proxy/Makefile [hunk 1]
```diff
@@ -3,14 +3,14 @@
 
 include $(TOPDIR)/rules.mk
 
-PKG_LICENSE:=GPL-3.0+
+PKG_LICENSE:=GPL-3.0-or-later
 PKG_MAINTAINER:=Stan Grishin <stangri@melmac.net>
 
 LUCI_TITLE:=HTTPS DNS Proxy Web UI
 LUCI_DESCRIPTION:=Provides Web UI for HTTPS DNS Proxy
-LUCI_DEPENDS:=+luci-compat +luci-mod-admin-full +https_dns_proxy
```
Comment (jow-) [inline applications/luci-app-https_dns_proxy/Makefile]: @stangri - I think you need to readd the `luci-compat` dependency, this is required if you use server side CBI rendering (like `Map()` or `Form()` in Lua files).

Comment (stangri) [reply applications/luci-app-https_dns_proxy/Makefile]: @jow- does the dependency on `luci-mod-admin-full` not satisfy `luci-compat`?

**Comment by feckert** [comment]
@stangri Sorry, but can you please fix the indentation

File: applications/luci-app-https_dns_proxy/Makefile (hunk 1 above)
Comment (stangri) [reply applications/luci-app-https_dns_proxy/Makefile]: @jow- could you please address the Q above as I may have to urgently send PRs for my other luci apps.

**Comment by stangri** [comment]
> @stangri Sorry, but can you please fix the indentation

I've tried changing the vscode settings to switch to tabs instead of double-spaces, looks like it didn't work. If I need to make the change in the Makefile, I'll run sed on lua files before I send the fix.

**Comment by stangri** [comment]
@jow- not sure if you're receiving notifications about my question on luci-compat. Please let me know,

File: applications/luci-app-https_dns_proxy/Makefile (hunk 1 above)
Comment (feckert) [reply applications/luci-app-https_dns_proxy/Makefile]: As I can see the dependency in the master branch is always `+luci-compat` and `+luci-mod-admin-full`

File: applications/luci-app-https_dns_proxy/luasrc/model/cbi/https_dns_proxy.lua [hunk 2]
```diff
@@ -1,144 +1,118 @@
...
--- prov:value("https://dns.digitale-gesellschaft.ch/dns-query?", "Digitale Gesellschaft (ch)")
-prov:value("https://doh.dns.sb/dns-query?", "DNS.SB")
-prov:value("https://dns.google.com/resolve?", "Google")
--- prov:value("https://odvr.nic.cz/doh?", "ODVR (nic.cz)")
--- prov:value("https://dns.quad9.net:5053/dns-query?", "Quad9 (Recommended)")
--- prov:value("https://dns9.quad9.net:5053/dns-query?", "Quad9 (Secured)")
--- prov:value("https://dns10.quad9.net:5053/dns-query?", "Quad9 (Unsecured)")
--- prov:value("https://dns11.quad9.net:5053/dns-query?", "Quad9 (Secured with ECS Support)")
-prov.default = "https://dns.google.com/resolve?"
+for filename in fs.dir(providers_dir) do
```
Comment (feckert) [inline applications/luci-app-https_dns_proxy/luasrc/model/cbi/https_dns_proxy.lua]: Sorry but could you fix the indentation

File: applications/luci-app-https_dns_proxy/luasrc/view/https_dns_proxy/js.htm [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    element.style.width = "16px";
+		element.innerHTML = '<img src="<%=resource%>/icons/loading.gif" alt="<%:Loading%>" width="16" height="16" style="vertical-align:middle" />';
+	}
+	else {
+    element.style.width = "0px";
+		element.innerHTML = '';
+	}
+}
+//]]>
+</script>
```
Comment (feckert) [inline applications/luci-app-https_dns_proxy/luasrc/view/https_dns_proxy/js.htm]: The same in this file.
Could you also fix the indentation into tab?

**Pull request closed without merging**